paddle/framework/operator.cc

```diff
@@ -411,7 +411,34 @@ void OperatorWithKernel::Run(const Scope& scope,
                  expected_kernel_key);
   }
 
-  kernel_iter->second->Compute(ctx);
+  if (actual_kernel_key == expected_kernel_key) {
+    kernel_iter->second->Compute(ctx);
+  } else {
+    Scope& op_scope = scope.NewScope();
+    auto input_vars = this->InputVars();
+    for (auto var_name : input_vars) {
+      op_scope.Var(var_name);
+    }
+
+    // TODO(qijun) get appropriate DeviceContext from DeviceContext pool
+    platform::DeviceContext* trans_dev_ctx = nullptr;
+
+    // TODO(qijun) get appropriate DataTransformFn from global map
+    using DataTransformFn = std::function<void(
+        const Variable& in, Variable* out, platform::DeviceContext* ctx)>;
```

Comment:
Variable = > Tensor?
We only use Tensor for kernel computing.

Reply:
Ok, I see.

Reply:
I am not quite sure why here need to be Variable but not Tensor, can you guys explain a bit?